mnemosyne/core/sync.py

```
@@ -875,194 +1457,446 @@ def pull_changes(
+            ).fetchone() is not None
+        if exists:
+            try:
+                from mnemosyne.core.beam import _wm_vec_delete
 
-        Events with an event_hash already present in the local log are
-        silently skipped (idempotent push).
-        """
-        cursor = self.conn.cursor()
+                _wm_vec_delete(self.conn, event.memory_id)
+            except Exception:
+                pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)